### PR TITLE
feat(modal-checkout): redesign variation selection

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

1205234045751551-as-1206185519218321

Redesign the modal checkout's variation selection according to 1205234045751551-as-1206122717080146.

The styles should still be updated to integrate with https://github.com/Automattic/newspack-plugin/pull/2810

image

image

image

### How to test the changes in this Pull Request:

1. Checkout this branch and https://github.com/Automattic/newspack-plugin/tree/epic/ras-acc
2. Create 2 variable subscription products in Woo, one with 4 variations, other with 2, as represented in the images above
3. Draft a page and add a Checkout Button block each of the created products
4. Make sure "Allow the reader to select the variation before checkout." in the block setting is toggled on, so the variation selection becomes available
5. Visit the page, click the button for the 4 variations, and confirm it renders in 3 columns as above
6. Switch to mobile viewport (below 600 pixels of width) and confirm the options stack as above
8. Click "Purchase" on any variation and confirm the modal checkout renders for the selected variation
9. Exit the modal checkout, click the button for the 2 variations, and confirm it renders in 2 columns as above
10. Click "Purchase" and confirm the modal checkout renders as expected

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by dkoo — APPROVED** [review]
Works well and looks great! Just one very minor edge case. Not sure if we can do much about this in a way that looks good, but if the price string is fairly long, it can easily exceed the width of the container in the three-column layout.

Screenshot 2024-01-15 at 5 32 44 PM

I'll consider this non-blocking as I think it's pretty unlikely to happen in live sites. The costliest product I've seen is about $2,500.00.

**Comment by miguelpeixe** [comment]
Good catch! It slipped me that @kevinzweerink requested to cut off the `.00` cents when the price is over 3 digits. Updated in 3c0981ad2b24be15b730e74c9db2917f4b2d5479.

**Review by dkoo — CHANGES_REQUESTED** [review]
> Good catch! It slipped me that @kevinzweerink requested to cut off the .00 cents when the price is 3+ digits. Updated in https://github.com/Automattic/newspack-blocks/commit/3c0981ad2b24be15b730e74c9db2917f4b2d5479.

Small nit with this change below.

File: includes/class-modal-checkout.php [hunk 1]
```diff
@@ -310,10 +310,28 @@ public static function render_modal_markup() {
...
+	 */
+	public static function update_subscriptions_product_price_string( $price_string ) {
+		$price_string = str_replace( ' / ', ' ' . __( 'per', 'newspack-blocks' ) . ' ', $price_string );
+		// For prices over 3 digits and 00 cents, remove the cents.
+		$pattern = '/\b\d{3,}\.\d{2}\b/';
+		preg_match( $pattern, $price_string, $matches );
+		if ( ! empty( $matches ) ) {
+			$price_string = preg_replace( $pattern, preg_replace( '/\.00$/', '', $matches[0] ), $price_string );
+		}
+		return $price_string;
```
[the remark this answers is not in the document]
Comment (miguelpeixe) [reply includes/class-modal-checkout.php:327]: Good call on using the decimal separator!

I'd rather keep treating this as a string since it's no longer a number at this point. Converting back to a number means re-applying the string logic from Woo for the separators and potentially other things, which is more than we should be doing here IMO.

[the remark this answers is not in the document]
Comment (miguelpeixe) [reply includes/class-modal-checkout.php]: Updated in 062284090c543f70134ddbdee9831a27448f0470

**Review by dkoo — APPROVED** [review]
Works now! Thanks for the update.

Screenshot 2024-01-19 at 4 20 17 PM
